### ISSUE TITLE
Can't view archived logs in UI

### ISSUE THREAD

**Issue by tim-sendible** [argoproj/argo-workflows#5983]
## Summary 

What happened/what you expected to happen?

User cannot view the logs in 3.1rc6 (also tested in rc5) despite the fact they are there.

- Enable log archiving.
- Install v3.1-rc6 and run a workflow. When completed look at the archived logs (user just gets the error "Artifact not found")
-  Downgrade to v3.0.4, view that completed workflow in the UI and look at the archived logs.

## Diagnostics

What Kubernetes provider are you using? EKS 1.19

What version of Argo Workflows are you running? 3.1-rc6

What executor are you running? Docker/K8SAPI/Kubelet/PNS/Emissary Docker

Did this work in a previous version? I.e. is it a regression? Works in 3.0.4

All the logs indicate that the logs have been archived, and there are no errors in logs when trying to view them.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
Can you confirm if the problem is that the logs are not viewable (UI issue) or they were never stored in the bucket?

**Comment by tim-sendible** [comment]
They are not viewable. They save fine.

**Comment by alexec** [comment]
@tim-sendible would you be willing to investigate, maybe with @dcherman help? The core team is short-staffed at the moment.

**Comment by dcherman** [comment]
@tim-sendible Are you getting any network errors / HTTP 500s at all?  I haven't had time to dig into anything really, but I did notice that with the code in master, the log viewer fails to load logs.  It's seemingly due to the following commit:

https://github.com/argoproj/argo-workflows/commit/284adfe16aeb11536a1c98f1956fdeb76dac4f1c

Where the main-logs is misidentified as an input artifact.  That HTTP 500 should almost certainly be a 404 as well since the artifact wasn't found.  Not sure if this is the same bug you're seeing or not, this commit was first released in 3.1.0-rc1 it looks like.

**Comment by tim-sendible** [comment]
@dcherman I have not seen any network errors at all. I will try and find time tomorrow to reinstall the latest RC and try again.

**Comment by alexec** [comment]
@dinever it seems your commit has introduced a regression. Would you please be able to investigate?

**Comment by dinever** [comment]
> @dinever it seems your commit has introduced a regression. Would you please be able to investigate?

Sure, will look into it

**Comment by svmaris** [comment]
Same problem here. The artifacts, including the logs, are in the S3-bucket but the UI shows a 502 Bad Gateway

error in the argo-server log:

```
time="2021-05-26T09:47:03.606Z" level=info msg="Download artifact" artifactName=main-logs isInput=true namespace=argo nodeId=terraform-tsvg5-3595155252 workflowName=terraform-tsvg5
2021/05/26 09:47:03 http: panic serving 100.96.15.6:41622: runtime error: invalid memory address or nil pointer dereference
goroutine 2284 [running]:
net/http.(*conn).serve.func1(0xc0006f6fa0)
	/usr/local/go/src/net/http/server.go:1801 +0x147
panic(0x2315e20, 0x43a8770)
	/usr/local/go/src/runtime/panic.go:975 +0x47a
github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.(*Inputs).GetArtifactByName(...)
	/go/src/github.com/argoproj/argo-workflows/pkg/apis/workflow/v1alpha1/workflow_types.go:2473
github.com/argoproj/argo-workflows/v3/server/artifacts.(*ArtifactServer).returnArtifact(0xc00020e300, 0x3407100, 0xc000d5cab0, 0x33fe5c0, 0xc0007947e0, 0xc000ec2500, 0xc0001f1c00, 0xc00061098a, 0x1a, 0xc0006109a5, ...)
	/go/src/github.com/argoproj/argo-workflows/server/artifacts/artifact_server.go:148 +0x118
github.com/argoproj/argo-workflows/v3/server/artifacts.(*ArtifactServer).getArtifact(0xc00020e300, 0x33fe5c0, 0xc0007947e0, 0xc000ec2500, 0x1)
	/go/src/github.com/argoproj/argo-workflows/server/artifacts/artifact_server.go:76 +0x5b1
github.com/argoproj/argo-workflows/v3/server/artifacts.(*ArtifactServer).GetInputArtifact(...)
	/go/src/github.com/argoproj/argo-workflows/server/artifacts/artifact_server.go:51
net/http.HandlerFunc.ServeHTTP(0xc000923910, 0x33fe5c0, 0xc0007947e0, 0xc000ec2500)
	/usr/local/go/src/net/http/server.go:2042 +0x44
net/http.(*ServeMux).ServeHTTP(0xc000920a40, 0x33fe5c0, 0xc0007947e0, 0xc000ec2500)
	/usr/local/go/src/net/http/server.go:2417 +0x1ad
net/http.serverHandler.ServeHTTP(0xc000794000, 0x33fe5c0, 0xc0007947e0, 0xc000ec2500)
	/usr/local/go/src/net/http/server.go:2843 +0xa3
net/http.(*conn).serve(0xc0006f6fa0, 0x3407040, 0xc0005f43c0)
	/usr/local/go/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2969 +0x36c
```

**Comment by svmaris** [comment]
Speaking of log artifacts: wouldn't it make more sense to automatically get the logs from the artifact repository when archiveLogs is enabled?

**Comment by svmaris** [comment]
@dinever I think the problem is here:

https://github.com/argoproj/argo-workflows/blob/c8df5eae262223e8d32d740f43b8a7c013aba4ce/ui/src/app/shared/services/workflows-service.ts#L201

The download-URL for the logs is treated as an "input" artifact instead of an "output" artifact. In fact, if you replace `/input-artifacts/` with `/artifacts/` the logs download just fine.

**Comment by dinever** [comment]
Thanks for the info @svmaris! I've opened #6019 for the fix.

**Comment by christopheblin** [comment]
@svmaris what is the 'archiveLogs' you are speaking about ? I'm looking into such a feature in #5996 but impossible to make it work ... (Pod failed: Error (exit code 1): key unsupported: cannot get key for artifact location, because it is invalid")

**Comment by svmaris** [comment]
@christopheblin I configured archiveLog in a ConfigMap named "artifact-repositories":

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  # if you want to use this config map by default - name it "artifact-repositories"
  name: artifact-repositories
  annotations:
    # v3.0 and after - if you want to use a specific key, put that's key into this annotation
    workflows.argoproj.io/default-artifact-repository: default-v1
data:
  default-v1: |
    archiveLogs: true
    s3:
      bucket: my-bucket-name
      endpoint: s3.amazonaws.com
      insecure: false
      accessKeySecret:
        name: argo-artifacts-s3-creds
        key: accesskey
      secretKeySecret:
        name: argo-artifacts-s3-creds
        key: secretkey
```

For an example using the default minio storage backend, see https://github.com/argoproj/argo-workflows/blob/master/manifests/quick-start/base/artifact-repositories-configmap.yaml

**Comment by tim-sendible** [comment]
I can confirm that this is resolved in 3.1.0-rc9. Thanks all.
